### PR TITLE
Feat/best score

### PR DESCRIPTION
**Pull request by RomainGuarinoni**
image

- handle the save of best score in the localStorage #51 
- Display the record at the end of the game #50

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -171,6 +177,10 @@ function startNewGame() {
   addTargetListener(currentTargets, COLORS);
 
   changeGameStatus(verifyButton, GameStatus.running);
+
+  console.log('Setting a new startDate');
+  runStart = new Date();
+  console.log(runStart);
 }
```
Comment (Guillaume-FAURE) [inline src/index.ts:185]: A lot of console.log, interesting for development but we don't want to keep them for the main

File: src/record.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
+type Category = {
+  duplicate: boolean;
+  nbTurns: number;
+  nbColors: number;
+  nbPossibilities: number;
+};
+
+export type Run = {
+  category: Category;
+  date: Date;
```
Comment (Guillaume-FAURE) [inline src/record.ts:10]: We also want the nb of turn the player did before win (not always the number of turn max of the category)

File: src/record.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ * @param {Run} run The user run
+ * @returns return a boolean that indicates whether the run is a new record or not
+ */
+export function isNewRecord(run: Run): boolean {
+  const previousRecord = getRecord(run.category);
+
+  if (!previousRecord) {
+    return true;
+  }
+
```
Comment (Guillaume-FAURE) [inline src/record.ts:70]: maybe see if we want to keep multiples records

**Review by Guillaume-FAURE — COMMENTED** [review]
some suggestions and console.log in index.ts useless

**Comment by RomainGuarinoni** [comment]
## Problems 

1. When close the endGame popup and re open it, the text change bcs we remake the calcul and the date change, fix that

File: src/index.ts (hunk 1 above)
Comment (RomainGuarinoni) [reply src/index.ts:185]: Done

File: src/record.ts (hunk 2 above)
Comment (RomainGuarinoni) [reply src/record.ts:10]: We could think about a better system to rank the run of the user based of the time took to finish the run and the lines took. Actually, it is just based on time.

File: src/dom-manipulation.ts [hunk 4]
```diff
@@ -184,27 +183,73 @@ export function getCurrentNumbersOfLine(): number {
...
+ * @param previousRecord The previous recorrd of this category
+ *
  */
-export function isGameFinish(verifyButton: HTMLButtonElement): boolean {
-  if (verifyButton.innerHTML == GameStatus.finish) return true;
+export function displayNewRecord(
+  p: HTMLParagraphElement,
+  newRecord: Run,
+  previousRecord: Run | null,
+) {
```
Comment (Guillaume-FAURE) [inline src/dom-manipulation.ts:243]: Isn't it interesting to add a function(): void cause it return nothing ?

**Review by Guillaume-FAURE — APPROVED** [review]
perfect, the boss

**Pull request merged**